src/cascade/input_data/configuration/form.py

```diff
@@ -0,0 +1,134 @@
+            elif self.density == "log_gaussian":
+                self.prior_object = priors.LogGaussianPrior(self.mean, self.std, self.eta, self.min, self.max)
+            elif self.density == "log_laplace":
+                self.prior_object = priors.LogLaplacePrior(self.mean, self.std, self.eta, self.min, self.max)
+            elif self.density == "log_students":
+                self.prior_object = priors.LogStudentsTPrior(self.mean, self.std, self.nu, self.eta, self.min, self.max)
+            else:
+                errors.append(f"Unknown density '{self.density}'")
+        except priors.PriorError as e:
+            errors.append(f"Parameters incompatible with density '{self.density}': {str(e)}")
+
+        return errors
+
+
+class SmoothingPriorGroup(Form):
+    dage = SmoothingPrior(name_field="prior_type")
+    dtime = SmoothingPrior(name_field="prior_type")
+    value = SmoothingPrior(name_field="prior_type")
+
+
+class Smoothing(Form):
+    rate = IntField()
+    age_grid = StringListField(constructor=float)
+    time_grid = StringListField(constructor=float)
+    default = SmoothingPriorGroup()
+    mulstd = SmoothingPriorGroup(nullable=True)
+    detail = FormList(SmoothingPrior, nullable=True)
+
+    custom_age_grid = Dummy()
+    custom_time_grid = Dummy()
+
+
+class Model(Form):
+    modelable_entity_id = IntField()
+    title = StrField()
+    description = StrField()
+    bundle_id = IntField(nullable=True)
+    drill = OptionField(["cascade", "drill"])
+    drill_location = IntField()
+    drill_sex = OptionField([1, 2], nullable=True)
+    default_age_grid = StringListField(constructor=float)
+    default_time_grid = StringListField(constructor=float)
+
+
+class Configuration(Form):
+    """ The root Form of the whole configuration tree.
+
+    Example:
+        >>> input_data = json.loads(json_blob)
+        >>> form = Configuration(input_data)
+        >>> errors = form.validate_and_normalize()
+        >>> if errors:
+                print(errors)
+                raise Exception("Woops")
+            else:
+                print(f"Ready to configure a model for {form.model.modelable_entity_id}")
+
+    """
+
+    model = Model()
+    gbd_round_id = IntField()
+    csmr_cod_output_version_id = IntField()
+    csmr_mortality_output_version_id = IntField()
+    location_set_version_id = IntField()
+    random_effect = FormList(Smoothing, nullable=True)
+    rate = FormList(Smoothing)
+
+    min_cv = FormList(Dummy)
+    min_cv_by_rate = FormList(Dummy)
+    re_bound_location = FormList(Dummy)
+    study_covariate = Dummy()
+    country_covariate = Dummy()
+    derivative_test = Dummy()
+    max_num_iter = Dummy()
+    print_level = Dummy()
+    accept_after_max_steps = Dummy()
+    tolerance = Dummy()
+    students_dof = Dummy()
+    log_students_dof = Dummy()
+    eta = Dummy()
+    data_eta_by_integrand = Dummy()
+    data_density_by_integrand = Dummy()
+    config_version = Dummy()
```

Comment:
This one's a dummy? We might want to read the config version early, which raises the question of whether we can version this set of parameters in any real way. Yes, we could. We'd have a root for each version. And we'd have to write code to promote settings from one root to fill out the settings in the next root. Seems unlikely we get to include that as a feature, but that's how it would work.

Reply:
Things I don't know what to do with are Dummy. That's true of this one. How do I use it? Aside from a scheme like you describe which I agree we aren't doing. Also, at the moment, I have no idea what EpiViz means by config version.